core/src/main/java/io/grpc/inprocess/InProcessChannelBuilder.java

```diff
@@ -246,6 +250,30 @@ public ScheduledExecutorService getScheduledExecutorService() {
       return timerService;
     }
 
+    @Override
+    public ClientTransportFactory withNewChannelCredential(ChannelCredentials channelCreds) {
+      checkNotNull(channelCreds, "channelCreds");
+      // InProcessChannel does not do any channel authentication, so it only supports insecure
```

Comment:
Let's not worry about in-process today, since it doesn't support channel credentials at all. I'd rather not define how it handles the credentials at this time, and it wouldn't be usable anyway from the LB because there's no name resolution support.

Reply:
Okay, throw UOE now.

Reply:
No, return `null`.

Reply:
> it wouldn't be usable anyway from the LB because there's no name resolution support.

I think it might be usable if the InProcessChannelBuilder is given a defaultLoadBalancingPolicy.

Reply:
It is possible to have a load balancer enabled with in-process. I was saying that LB couldn't use the createResolvingOobChannelBuilder and get anything useful to happen.

The API being added here uses a target string which requires a name resolver. In-process is only able to create connections to `InProcessSocketAddress`. So every name resolver today would basically break it and if there were some other potential name resolver that produced InProcessSocketAddress that name resolver would be incompatible with all the other transports.

Reply:
Done returning null.